src/main/java/com/puppycrawl/tools/checkstyle/ThreadModeSettings.java

```diff
@@ -25,6 +25,7 @@
  * Thread mode settings for the checkstyle modules.
  *
  * @noinspection SerializableHasSerializationMethods
+ * @noinspectionreason we do not serialize class
  */
 public class ThreadModeSettings implements Serializable {
```

Comment:
this class is not API - lets fix it.

Reply:
Done

Reply:
See https://dev.azure.com/romanivanovjr/a35b6275-caac-4633-b371-79a5ab0f70c1/_apis/build/builds/9482/logs/100, this would be a significant change if we decide to do this.